### ISSUE TITLE
[NODE] GUI Blank Screen while TOML job is executing

### ISSUE THREAD

**Issue by pappas999** [smartcontractkit/chainlink#5263]
**Description**
After creating a directrequest job, either a normal HTTP GET job, or one that uses a bridge/external adapter, the job creates successfully. However when a consumer contract then goes to use the job and you go to refresh the node GUI screen to see the job executing in real-time, the GUI returns a blank screen. Once the job has completed and finished executing (whether successfully or failure), you can refresh the screen again and the GUI will resume, and you can see the job output

**Basic Information**
[replace this line with basic information about the issue you are experiencing, including but not limited to all relevant logs and any other relevant information, such as if you are using a Docker container to run the node, job specification, oracle contract address, transaction IDs, etc.]

- Network: Kovan
- Blockchain Client: Infura
- Go Version: 1.17
- Operating System: MacOs
- Node Version: 1.0.0

**Steps to Reproduce**
- Spin up v.1.0.0 node, deploy oracle.sol and set fulfill permission to the node address
- Create a new directrequest TOML job, as per the example below
- Create a new consumer contract in remix that uses the created job and deployed oracle contract, as per the example below
- Execute the requestEthereumPrice function on the deployed contract, passing in the oracle contract address and jobId
- Go back to the GUI and try go to the job runs screen. While the job is executing the GUI will return a blank screen. Even if you go to another page (like the node homepage), nothing will show up on the GUI
- Wait 30 seconds until the job has completed, then refresh the GUI screen, you should now see the GUI again

**Additional Information**

Error message in node console:
```
2021-10-20T13:00:34Z [DEBUG] HeadTracker: Received new head #27835174 (0x1a8bb26) headtracker/head_listener.go:147 blockHash=0x4a93e3c4929f8020e72fd22b669b823f2460d53d1f4c17195e66f05c14b89109 blockHeight=27835174 logger=head_tracker parentHeadHash=0x0cde80a95131ae26500322a8487da6fa3e1a2a54f87ed4e30f70f0c88f0db7fa 
2021-10-20T13:00:35Z [INFO]  New ETH balance for 0xF8DaCFFe864D900C5Dd99576578Ee066086D5a70: 0.999247024000000000 services/balance_monitor.go:166 address=0xF8DaCFFe864D900C5Dd99576578Ee066086D5a70 chainId=42 ethBalance=0.999247024000000000 id=balance_log weiBalance=999247024000000000 
2021-10-20T13:00:36Z [INFO]  GET /v2/jobs                                       web/router.go:430       clientIP=::1 latency=9.485323ms method=GET path=/v2/jobs servedAt=2021-10-20 23:30:36 status=200 
2021-10-20T13:00:37Z [ERROR] PipelineRunResource: unable to process output type <nil> presenters/pipeline_run.go:131 stacktrace=github.com/smartcontractkit/chainlink/core/web/presenters.NewPipelineRunResources
	/Users/pappas99/GitHub/chainlink/core/web/presenters/pipeline_run.go:131
github.com/smartcontractkit/chainlink/core/web.(*PipelineRunsController).Index
	/Users/pappas99/GitHub/chainlink/core/web/pipeline_runs_controller.go:60
github.com/smartcontractkit/chainlink/core/web.paginatedRequest.func1
	/Users/pappas99/GitHub/chainlink/core/web/helpers.go:88
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/smartcontractkit/chainlink/core/web.RequireAuth.func1
	/Users/pappas99/GitHub/chainlink/core/web/authentication.go:131
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/gin-gonic/contrib/sessions.Sessions.func1
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/contrib@v0.0.0-20190526021735-7fb7810ed2a0/sessions/sessions.go:65
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/ulule/limiter/drivers/middleware/gin.(*Middleware).Handle
	/Users/pappas99/go/pkg/mod/github.com/ulule/limiter@v0.0.0-20190417201358-7873d115fc4e/drivers/middleware/gin/middleware.go:57
github.com/ulule/limiter/drivers/middleware/gin.NewMiddleware.func1
	/Users/pappas99/go/pkg/mod/github.com/ulule/limiter@v0.0.0-20190417201358-7873d115fc4e/drivers/middleware/gin/middleware.go:33
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/Depado/ginprom.(*Prometheus).Instrument.func1
	/Users/pappas99/go/pkg/mod/github.com/!depado/ginprom@v1.2.1-0.20200115153638-53bbba851bd8/prom.go:215
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/recovery.go:99
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/smartcontractkit/chainlink/core/web.loggerFunc.func1
	/Users/pappas99/GitHub/chainlink/core/web/router.go:427
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/gin-contrib/size.RequestSizeLimiter.func1
	/Users/pappas99/go/pkg/mod/github.com/gin-contrib/size@v0.0.0-20190528085907-355431950c57/size.go:88
github.com/gin-gonic/gin.(*Context).Next
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/context.go:165
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/gin.go:489
github.com/gin-gonic/gin.(*Engine).ServeHTTP
	/Users/pappas99/go/pkg/mod/github.com/gin-gonic/gin@v1.7.2/gin.go:445
net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:2878
net/http.(*conn).serve
	/usr/local/go/src/net/http/server.go:1929 
2021-10-20T13:00:37Z [INFO]  GET /v2/jobs/15/runs                               web/router.go:430       clientIP=::1 latency=6.293717ms method=GET path=/v2/jobs/15/runs query=page=1&size=5 servedAt=2021-10-20 23:30:37 status=200 
2021-10-20T13:00:37Z [INFO]  GET /v2/jobs/15                                    web/router.go:430       clientIP=::1 latency=8.556668ms method=GET path=/v2/jobs/15 servedAt=2021-10-20 23:30:37 status=200 
```

This is semi related to other blank screen GUI issues https://github.com/smartcontractkit/chainlink/issues/4964 and https://github.com/smartcontractkit/chainlink/issues/4882

TOML spec to reproduce issue:

```
type = "directrequest"
schemaVersion = 1
name = "Get > Uint256-2"
contractAddress = "0xA74F1E1Bb6204B9397Dac33AE970E68F8aBC7651"
maxTaskDuration = "0s"
observationSource = """
    decode_log   [type=ethabidecodelog
                  abi="OracleRequest(bytes32 indexed specId, address requester, bytes32 requestId, uint256 payment, address callbackAddr, bytes4 callbackFunctionId, uint256 cancelExpiration, uint256 dataVersion, bytes data)"
                  data="$(jobRun.logData)"
                  topics="$(jobRun.logTopics)"]

    decode_cbor  [type=cborparse data="$(decode_log.data)"]
    fetch        [type=http method=GET url="$(decode_cbor.get)"]
    parse        [type=jsonparse path="$(decode_cbor.path)" data="$(fetch)"]
    multiply     [type=multiply input="$(parse)" times=100]     
    encode_data  [type=ethabiencode abi="(uint256 value)" data=<{ "value": $(multiply) }>]
    encode_tx    [type=ethabiencode
                  abi="fulfillOracleRequest(bytes32 requestId, uint256 payment, address callbackAddress, bytes4 callbackFunctionId, uint256 expiration, bytes32 data)"
                  data=<{"requestId": $(decode_log.requestId), "payment": $(decode_log.payment), "callbackAddress": $(decode_log.callbackAddr), "callbackFunctionId": $(decode_log.callbackFunctionId), "expiration": $(decode_log.cancelExpiration), "data": $(encode_data)}>
                 ]
    submit_tx    [type=ethtx to="0xA74F1E1Bb6204B9397Dac33AE970E68F8aBC7651" data="$(encode_tx)"]

    decode_log -> decode_cbor -> fetch -> parse -> multiply -> encode_data -> encode_tx -> submit_tx
"""


```
[Consuming Solidity contract](https://remix.ethereum.org/#url=https://deploy-preview-427--dreamy-villani-0e9e5c.netlify.app/samples/APIRequests/ATestnetConsumer.sol&optimize=false&runs=200&evmVersion=null&version=soljson-v0.8.7+commit.e28d00a7.js) to reproduce issue

**Comment by nemani** [comment]
I ran into this issue as well, in 2 different ways.

- If there is a syntax error in the job spec which causes the building of graph to fail then the js halts at the error, and shows a blank screen when you try to create a new job
- If the last task in a job pipeline returns null then the UI doenst know how to process it when you try to visit the specific job page in the UI.

**Comment by furfio** [comment]
> thanks for the feedback @dwightjl and @ebadiere . The front-end team will take a look at this hopefully soon

hi, I still meet with this blank screen problems, it seems to be caused by pending transaction, actually, I am a new comer for chainlink, I follow your video https://www.youtube.com/watch?v=DJ-EYU-7f0U, but I just cannot see runs in my job after I click my requestGame button,  I also follow zak's video, but every transaction cost almost 20minutes to complete during which I got a just blank screen.  The network is just so bad in kovan, is there any solutions?

**Comment by furfio** [comment]
> 
I have list my question here:
https://ethereum.stackexchange.com/questions/117333/chainlink-hackathon-building-and-using-external-adapters

**Comment by eduardotang** [comment]
the UI shows blank whenever visiting the route /v2/pipeline/runs or /jobs/1/runs  
there is an exception in getting the pipelinerunresource 

as you can see from this line
[ERROR] PipelineRunResource: unable to process output type <nil> presenters/pipeline_run.go:131

when having null in either errors or outputs or both in the pipeline_runs table , it has problem either handling nil or serializing "jsonb-typed" null value
so it comes to either when constructing in "NewPipelineRunResource" or the model "pipeline.Run"

if you manually go into the DB and change these 2 columns to [null] or [""]
the exception goes away and the UI will display

using docker image    **smartcontract/chainlink:1.0.1**

**Comment by KuphJr** [comment]
I can confirm I am having the exact same issue with my Chainlink node.  I am also using Docker image **smatcontract/chainlink:1.0.1**

**Comment by gen4sp** [comment]
My way to deal with issue: 
chrome dev tools > application > local storage >  clear the saved task

**Comment by KuphJr** [comment]
> My way to deal with issue: chrome dev tools > application > local storage > clear the saved task

I have tried doing that, but the same problem occurs.  Basically, as long as the job is still in progress, every time you refresh the webpage 'https://localhost:6688/jobs/JOB_NUMBER_HERE', the webpage is blank.  As soon as the job completes, the webpage operates normally upon refresh.

This seems to occur regardless if all the local storage is cleared.  I even tried to log in on a completely different web browser that I had not previously used and it is the same.

**Comment by dmpierre** [comment]
Hey :)

I was confronted with the same issue for both jobs in the [fulfilling requests](https://docs.chain.link/docs/fulfilling-requests/) documentation section using image `smartcontract/chainlink:1.0.1`. 

My workaround was to downgrade to image `smartcontract/chainlink:0.10.15`.

**Comment by pappas999** [comment]
thanks for raising @dmpierre , can confirm I get the behaviour with 1.0.1 as well still. Will push back to the team to see where this is at

**Comment by PatrickAlphaC** [comment]
It looks like this is fixed in 1.1.1 yes?

**Comment by ebadiere** [comment]
It looks like it is fixed in my 1.1.1 installation.
